src/main/java/edu/utdallas/davisbase/storage/IndexPage.java

```diff
@@ -0,0 +1,497 @@
+			splitLeafNodeData(file, pageNo, parentPageNo, newSiblingPageNo);
+			setParentRightChildPointer(file, parentPageNo, newSiblingPageNo);
+			setRightSiblingPointer(file, newSiblingPageNo, getRightSiblingPointer(file, pageNo));
+			setRightSiblingPointer(file, pageNo, newSiblingPageNo);
+
+		} catch (Exception e) {
+		}
+		return -1;
+	}
+
+	public static void setParent(RandomAccessFile file, int pageNo, int parentPageNo) {
+		long seekParentByte = convertPageNoToFileOffset(pageNo) + PAGE_OFFSET_OF_CELL_PARENT_PAGE_NO;
+		try {
+			file.seek(seekParentByte);
+			file.writeInt(parentPageNo);
+		} catch (Exception e) {
+
+		}
+	}
+
+	public static void splitInteriorPage(RandomAccessFile file, int pageNo) throws IOException {
+		sortKeys(file, pageNo);
+		int newSiblingPageNo = AddInteriorPage(file);
+		boolean rootflag = CheckifRootNode(file, pageNo);
+		int parentPageNo;
+		try {
+			if (rootflag) {
+				parentPageNo = AddInteriorPage(file);
+			} else {
+				parentPageNo = getParent(file, pageNo);
+			}
+			splitInteriorNodeData(file, pageNo, parentPageNo, newSiblingPageNo);
+			setParentRightChildPointer(file, parentPageNo, newSiblingPageNo);
+
+		} catch (Exception e) {
+
+		}
+	}
+
+	private static void splitInteriorNodeData(RandomAccessFile file, int pageNo, int parentPageNo, int siblingPageNo) {
+		// TODO make sure to update left child pointers
+		// TODO Auto-generated method stub
+
+		try {
+			long bigPageOffset = convertPageNoToFileOffset(pageNo);
+			file.seek(bigPageOffset + PAGE_OFFSET_OF_CELL_COUNT);
+			int noOfrecordsInBigPage = file.readShort();
+			int splitIndex = (int) Math.ceil(noOfrecordsInBigPage / 2); // this record goes into the parent.
+			// add split index + 1 th element in the array to parent.
+			addMiddleElementFromChildtoParent(file, splitIndex, pageNo, parentPageNo);
+
+			// add element above split index+1 from big page to new page.
+			addElementsAboveIToSibling(file, splitIndex, pageNo, siblingPageNo);
+
+			// remove elements from split index +1 th element.
+			removeElementsAfterIthPosition(file, splitIndex, pageNo);
+
+		} catch (IOException e) {
+		}
+
+	}
+
+	private static void setRightSiblingPointer(RandomAccessFile file, int pageNo, int siblingPageNo) {
+		try {
+			long currentPageOffset = convertPageNoToFileOffset(pageNo);
+			file.seek(currentPageOffset + PAGE_OFFSET_OF_RIGHTMOST_PAGE_NO);
+			file.writeInt(siblingPageNo);
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+
+	}
+
+	private static int getRightSiblingPointer(RandomAccessFile file, int pageNo) throws IOException {
+		long currentPageOffset = convertPageNoToFileOffset(pageNo);
+		file.seek(currentPageOffset + PAGE_OFFSET_OF_RIGHTMOST_PAGE_NO);
+		return file.readInt();
+
+	}
+
+	private static void setParentRightChildPointer(RandomAccessFile file, int parentPageNo, int newleafPageNo) {
+		// TODO Auto-generated method stub
+		setRightSiblingPointer(file, parentPageNo, newleafPageNo);
```

Comment:
Please add a comment explaining why `setRightSiblingPointer(*)` can be used to `setParentRightChildPointer(*)`, since that seems like two different things entirely.